src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -476,92 +245,215 @@ HTTPProxy::_handleCreateSchema(sh::dto::CreateSchemaRequest&& request) {
+    return MakeHTTPResponse<shd::TxnEndResponse>(sh::Statuses::S501_Not_Implemented("txn end not implemented"), shd::TxnEndResponse{});
+}
+
+seastar::future<std::tuple<sh::Status, shd::GetSchemaResponse>>
+HTTPProxy::_handleGetSchema(shd::GetSchemaRequest&& request) {
+    K2LOG_D(log::httpproxy, "Received get schema request {}", request);
+    return MakeHTTPResponse<shd::GetSchemaResponse>(sh::Statuses::S501_Not_Implemented("get schema not implemented"), shd::GetSchemaResponse{});
+}
 
-    return seastar::when_all_succeed(_endFuts.begin(), _endFuts.end());
+seastar::future<std::tuple<sh::Status, shd::CreateQueryResponse>>
+HTTPProxy::_handleCreateQuery(shd::CreateQueryRequest&& request) {
+    K2LOG_D(log::httpproxy, "Received create query request {}", request);
+    return MakeHTTPResponse<shd::CreateQueryResponse>(sh::Statuses::S501_Not_Implemented("create query not implemented"), shd::CreateQueryResponse{});
+}
+
+seastar::future<std::tuple<Status, std::shared_ptr<dto::Schema>, std::shared_ptr<shd::Schema>>>
+HTTPProxy::_getSchemas(sh::String cname, sh::String sname, int64_t sversion) {
+    return _client.getSchema(cname, sname, sversion)
+        .then([this, cname=std::move(cname)](GetSchemaResult&& result) mutable {
+            if (!result.status.is2xxOK()) {
+                return seastar::make_ready_future<std::tuple<Status, std::shared_ptr<dto::Schema>, std::shared_ptr<shd::Schema>>>(std::move(result.status), std::shared_ptr<dto::Schema>(), std::shared_ptr<shd::Schema>());
+            }
+            // create the nested maps as needed - we have a schema
+            auto& shdSchemaPtr = _shdSchemas[cname][result.schema->name][result.schema->version];
+            if (!shdSchemaPtr) {
+                std::vector<shd::SchemaField> shdfields;
+
+                for (auto& f : result.schema->fields) {
+                    shdfields.push_back(shd::SchemaField{
+                        .type = static_cast<shd::FieldType>(f.type),
+                        .name = sh::String(f.name.data(), f.name.size()),
+                        .descending = f.descending,
+                        .nullLast = f.nullLast});
+                }
+                shd::Schema* schema  = new shd::Schema{
+                    .name = result.schema->name,
+                    .version = result.schema->version,
+                    .fields = std::move(shdfields),
+                    .partitionKeyFields = result.schema->partitionKeyFields,
+                    .rangeKeyFields = result.schema->rangeKeyFields};
+                shdSchemaPtr.reset(schema);
+            }
+            return seastar::make_ready_future<std::tuple<Status, std::shared_ptr<dto::Schema>, std::shared_ptr<shd::Schema>>>(std::move(result.status), std::move(result.schema), std::move(shdSchemaPtr));
+        });
+}
+
+HTTPProxy::HTTPProxy() : _client(K23SIClientConfig()) {
+}
+
+seastar::future<> HTTPProxy::gracefulStop() {
+    _txns.clear();
```

Comment:
Should it abort the txns like before?

Reply:
This functionality should be implemented by the k23si client itself - when it shuts down, it should cleanup any in-progress txns. There is no reason why the user of the client library has to worry about it.